neo/rawio/plexonrawio.py

```diff
@@ -264,22 +267,49 @@ def _parse_header(self):
             signal_streams = np.array([], dtype=_signal_stream_dtype)
 
         else:
-            # detect groups (aka streams)
-            all_sig_length = all_sig_length = np.array(all_sig_length)
-            groups = set(zip(sig_channels["sampling_rate"], all_sig_length))
+            # Detect streams
+            all_sig_length = np.asarray(all_sig_length)
+
+            # names are WB{number}, FPX{number}, SPKCX{number}, AI{number}, etc
+            pattern = r"^\D+"  # Match any non-digit character at the beginning of the string
+            channels_prefixes = np.asarray([re.match(pattern, name).group(0) for name in sig_channels["name"]])
+            buffer_stream_groups = set(zip(channels_prefixes, sig_channels["sampling_rate"], all_sig_length))
+
+            # There are explanations of the streams bassed on channel names
+            # provided by a Plexon Engineer, see here:
+            # https://github.com/NeuralEnsemble/python-neo/pull/1495#issuecomment-2184256894
+            channel_prefix_to_stream_name = {
+                "WB": "WB-Wideband",
+                "FP": "FPl-Low Pass Filtered ",
+                "SP": "SPKC-High Pass Filtered",
+                "AI": "AI-Auxiliary Input",
+            }
+
+            # Using a mapping to ensure consistent order of stream_index
+            channel_prefix_to_stream_id = {
+                "WB": "0",
+                "FP": "1",
+                "SP": "2",
+                "AI": "3",
+            }
 
             signal_streams = []
             self._signal_length = {}
             self._sig_sampling_rate = {}
-            for stream_index, (sr, length) in enumerate(groups):
-                stream_id = str(stream_index)
+
+            for stream_index, (channel_prefix, sr, length) in enumerate(buffer_stream_groups):
+                # The users of plexon can modify the channel names, is not common but in that case
+                # We assign the channel_prefix both as stream_name and stream_id
```

Comment:
What does this comment mean? 

So they can re-name a channel to not be WB, FB, SP, AI etc... What part can they actually rename? Do we need to protect against that better or could we make this comment clearer to what the renaming actually entails?

Reply:
The can modify the prefix. For example, in the test file that we had before I added the new one, the channel is named V1. 

In that case I am adding V as both the stream name and id. This can't be generalized but the user changed the prefix for some reason so they should know what it means in neo when they see the header and in spikeinterface when they get the assertion that the stream was not found.

Reply:
Okay I think I see what you mean! Thanks.

Reply:
Great! Let me know if there is an improvement to the comment that I could make.

Reply:
I guess it would make a little more sense in the comment to clarify channel name vs channel prefix. In the first part you say channel names, then you say we take the channel prefix. So making it clear: can the user modify the prefix, all of the name, the numbering of the channel.